File: oj-web/src/enums/oj/code/CodeMonitorStatus.ts
import type { BaseEnum } from '@/enums/base';

interface CodeMonitorStatusEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

export const CodeMonitorStatus: { [key: string]: CodeMonitorStatusEnum } = {
  MONITOR_PUSH: {
    code: 'MonitorPush',
    name: '监控推送',
    tooltip: '监控端（学生）推送代码',
    text: '监控端（学生）推送代码'
  },
  COVERED_PUSH: {
    code: 'CoveredPush',
    name: '覆盖推送',
    tooltip: '覆盖端（老师）推送代码',
    text: '覆盖端（老师）推送代码'
  },
  MONITOR_PULL: {
    code: 'MonitorPull',
    name: '监控拉取',
    tooltip: '监控端（学生）拉取代码',
    text: '监控端（学生）拉取代码'
  },
  MONITOR_REJECT: {
    code: 'MonitorReject',
    name: '监控拒绝',
    tooltip: '监控端（学生）拒绝监控',
    text: '监控端（学生）拒绝监控'
  }
};

export function matchCode(code: string): CodeMonitorStatusEnum | null {
  return Object.values(CodeMonitorStatus).find(item => item.code === code) || null;
} 

File: glowxq-api/business/business-oj/src/main/java/com/glowxq/oj/code/enums/CodeMonitorStatus.java
package com.glowxq.oj.code.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum CodeMonitorStatus implements BaseType {
    MonitorPush("MonitorPush", "监控端（学生）推送代码"),
    CoveredPush("CoveredPush", "覆盖端（老师）推送代码"),
    MonitorPull("MonitorPull", "监控端（学生）拉取代码"),
    MonitorReject("MonitorReject", "监控端（学生）拒绝代码"),;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static CodeMonitorStatus matchCode(String code) {
        for (CodeMonitorStatus pushStatus : CodeMonitorStatus.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}
